schism/compiler.ss

```diff
@@ -642,6 +722,133 @@
         '()
         (cons 'i32 (args->types (cdr args)))))
 
+  ;; ====================== ;;
+  ;; Simplification         ;;
+  ;; ====================== ;;
+
+  ;; The parsing phase introduces eta-expanded lambdas for top-levels
+  ;; and intrinsics, and canonicalizes "if" tests to be relop
+  ;; intrinsics.  This pass simplifies away the introduced complexity
+  ;; where it's not needed.
```

Comment:
Does this only simplify the introduced eta expansions, or does it simplify other cases as well? Like, if I write `((lambda (a b) (+ a b)) 1 2)`, does that also get simplified to `(+ 1 2)`?

Reply:
It will also beta-reduce source terms, yes.  Of course it's what you want in all cases except when testing ;)  So it could be worth it to take a look at tests to verify that this patch doesn't reduce coverage.

Reply:
Off hand, I don't know of any tests that rely on this behavior. Even in that case, I think reducing the expression is probably okay. In my mind, tests usually make sure the behavior is okay but aren't terribly concerned with whether the behavior happens at compile time or run time.